peerconnection_test.go

```diff
@@ -386,6 +388,14 @@ func TestPeerConnection_EventHandlers(t *testing.T) {
 			wg.Done()
 		})
 	})
+	pcOffer.OnConnectionStateChange(func(PeerConnectionState) {
```

Comment:
```suggestion
	pcOffer.OnConnectionStateChange(func(s PeerConnectionState) {
              if s2 := pcOffer.ConnectionState(); s != s2 {
                      t.Errorf("State in callback argument is different from ConnectionState(): arg=%s, state=%s", s.String(), s2.String())
              }
```
results
```
$ go test . -run TestPeerConnection_EventHandlers
--- FAIL: TestPeerConnection_EventHandlers (2.01s)
    peerconnection_test.go:393: State in callback argument is different from ConnectionState(): arg=connected, state=new
    peerconnection_test.go:393: State in callback argument is different from ConnectionState(): arg=connected, state=new
FAIL
FAIL	github.com/pion/webrtc/v2	2.037s
```
`pc.connectionState` is not updated and `pc.onConnectionStateChangeHandler` called duplicately.

Reply:
Great catch! Thanks for the test suggestion, added it